libs/messenger/src/messenger/messenger_prototype.cpp

```diff
@@ -0,0 +1,195 @@
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "messenger/messenger_prototype.hpp"
+
+#include <cassert>
+
+namespace fetch {
+namespace messenger {
+
+MessengerPrototype::MessengerPrototype(muddle::MuddlePtr &muddle, Addresses node_addresses)
+  : endpoint_{muddle->GetEndpoint()}
+  , rpc_client_{"Messenger", endpoint_, SERVICE_MESSENGER, CHANNEL_RPC}
+  , message_subscription_{endpoint_.Subscribe(SERVICE_MESSENGER, CHANNEL_MESSENGER_MESSAGE)}
+  , node_addresses_{std::move(node_addresses)}
+{
+  //    rpc_server_ = std::make_shared<Server>(endpoint_, SERVICE_MESSENGER, CHANNEL_RPC);
```

Comment:
This is deliberately dead as it will be activated in one of the next iterations of the code. I would prefer it to stay, although I could also just create and issue and keep the code snippet there.